processor/tailsamplingprocessor/sampling/string_tag_filter.go

```diff
@@ -90,3 +124,44 @@ func (saf *stringAttributeFilter) Evaluate(_ pdata.TraceID, trace *TraceData) (D
 	}
 	return NotSampled, nil
 }
+
+// addFilters compiles all the given filters and stores them as regexes.
+// All regexes are automatically anchored to enforce full string matches.
+func addFilters(exprs []string) []*regexp.Regexp {
+	list := make([]*regexp.Regexp, 0, len(exprs))
+	for _, entry := range exprs {
+		rule := regexp.MustCompile(entry)
+		list = append(list, rule)
+	}
+	return list
+}
+
+// Matches returns true if the given string matches any of the string attribute filters
+func (saf *stringAttributeFilter) matches(toMatch string) bool {
+	if saf.regexStrEnabled {
+		return saf.matchRegexStr(toMatch)
+	}
+
+	if _, ok := saf.values[toMatch]; ok {
+		return true
+	}
+	return false
+}
+
+// Matches returns true if the given string matches the regex rules defined in string attribute filters
```

Comment:
`// matchRegexStr returns ...` should be better, though the method is not exported so it won't show up in godoc.

Reply:
Updated, Thx